code/game/objects/structures/props/beam_prism.dm

```diff
@@ -0,0 +1,218 @@
+
+/obj/structure/prop/prism
+	name = "prismatic turret"
+	desc = "A raised, externally powered 'turret'. It seems to have a massive crystal ring around its base."
+	description_info = "This device is capable of redirecting any beam projectile."
+	icon = 'icons/obj/props/prism.dmi'
+	icon_state = "prism"
+	density = TRUE
+	anchored = TRUE
+
+	layer = 3.1					//Layer over projectiles.
+	plane = -10					//Layer over projectiles.
+
+	var/rotation_lock = 0		// Can you rotate the prism at all?
+	var/free_rotate = 1			// Does the prism rotate in any direction, or only in the eight standard compass directions?
+	var/external_control_lock = 0	// Does the prism only rotate from the controls of an external switch?
+	var/degrees_from_north = 0	// How far is it rotated clockwise?
+	var/compass_directions = list("North" = 0, "South" = 180, "East" = 90, "West" = 270, "Northwest" = 315, "Northeast" = 45, "Southeast" = 135, "Southwest" = 225)
+	var/interaction_sound = 'sound/mecha/mechmove04.ogg'
+
+	var/redirect_type = /obj/item/projectile/beam
+
+	var/dialID = null
+	var/obj/structure/prop/prismcontrol/remote_dial = null
+
+	interaction_message = "<span class='notice'>The prismatic turret seems to be able to rotate.</span>"
+
+/obj/structure/prop/prism/initialize()
+	if(degrees_from_north)
+		animate(src, transform = turn(NORTH, degrees_from_north), time = 3)
+
+/obj/structure/prop/prism/Destroy()
+	if(remote_dial)
+		remote_dial.my_turrets -= src
+		remote_dial = null
+	..()
+
+/obj/structure/prop/prism/proc/reset_rotation()
+	var/degrees_to_rotate = -1 * degrees_from_north
+	animate(src, transform = turn(src.transform, degrees_to_rotate), time = 2)
+
+/obj/structure/prop/prism/attack_hand(mob/living/user)
+	..()
+
+	if(rotation_lock)
+		to_chat(user, "<span class='warning'>\The [src] is locked at its current bearing.</span>")
+		return
+	if(external_control_lock)
+		to_chat(user, "<span class='warning'>\The [src]'s motors resist your efforts to rotate it. You may need to find some form of controller.</span>")
+		return
+
+	var/confirm = input("Do you want to try to rotate \the [src]?", "[name]") in list("Yes", "No")
+	if(confirm == "No")
+		visible_message(\
+			"<span class='notice'>[user.name] decides not to try turning \the [src].</span>",\
+			"<span class='notice'>You decide not to try turning \the [src].</span>")
+		return
+
+	var/new_bearing
+	if(free_rotate)
+		new_bearing = input("What bearing do you want to rotate \the [src] to?", "[name]") as num
+		new_bearing = round(new_bearing)
+		if(new_bearing <= -1 || new_bearing > 360)
+			to_chat(user, "<span class='warning'>Rotating \the [src] [new_bearing] degrees would be a waste of time.</span>")
+			return
+	else
+		var/choice = input("What point do you want to set \the [src] to?", "[name]") as null|anything in compass_directions
+		new_bearing = round(compass_directions[choice])
+
+	var/rotate_degrees = new_bearing - degrees_from_north
+
+	if(new_bearing == 360) // Weird artifact.
```

Comment:
In case you didn't test values over 360 and don't want it turning more than a full revolution in one action for whatever reason, `new_bearing %= 360` will force it to be on [0, 359]

Reply:
I want 360 to be an applicable input, and reset to 0 as it is. 360 when input into the rotation method for some reason points it randomly one degree to the left or right, but 0 doesn't.